sdk/core/Azure.Core/src/Shared/AzureSasCredentialSynchronousPolicy.cs

```diff
@@ -31,7 +31,24 @@ public override void OnSendingRequest(HttpMessage message)
             }
             if (!query.Contains(signature))
             {
-                query = string.IsNullOrEmpty(query) ? '?' + signature : query + '&' + signature;
+                bool setSignature = false;
+                // check if the signature has updated since we started processing this message
+                if (message.ProcessingStartTime.Ticks < _credential.SignatureUpdated)
+                {
+                    foreach (string previousSig in _credential.SignatureHistory.ToArray())
+                    {
+                        if (query.Contains(previousSig))
+                        {
+                            query = query.Replace(previousSig, signature);
+                            setSignature = true;
+                            break;
+                        }
+                    }
+                }
+                if (!setSignature)
```

Comment:
How could this occur? I thought the library would need to set it initially?

Reply:
- Request sent
 - SasCredential.Update called
 - Request fails and retries
 - Second request (first retry) goes through this path because [this check](https://github.com/Azure/azure-sdk-for-net/pull/32888/files/ae1a7c48a39a1bae1063b80353db143adb3334c0#diff-b60f5438f5cab373d46faf5bdaafbf4aae06f264d8c9de9004dbeae4e33b1c1fR32) doesn't detect the current signature